### PR TITLE
[AMORO-2651] Enable checkstyle when building project

### LINKED ISSUE #2651: [Improvement]: Add checkstyle check for wildcard import
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

throw exception if there is any wildcard import when building

### How should we improve?

We can add checkstyle plugin for this.

Why not add new spotless rule for this?
1. currently spotless can only use google java format 1.7(because this is the highest version support JDK8)
2. google java format did not ban wildcard import in 1.7
3. spotless did not support custom configure for maven(only supported in gradle)

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by klion26**
## Why are the changes needed?

Close #2651.

## Brief change log

-

## How was this patch tested?

- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1040,6 +1040,28 @@
                         </execution>
                     </executions>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-checkstyle-plugin</artifactId>
+                    <version>3.3.1</version>
```
Comment (zhoujinsong) [inline pom.xml:1046]: How about defining a property like `maven-checkstyle-plugin.version`?

File: tools/maven/checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+            <property name="illegalPattern" value="true"/>
+            <property name="message"
+                      value="Do not use Guava's Preconditions"/>
+        </module>
+
+        <module name="Regexp">
+            <property name="format" value="import static com\.google\.common\.base\.Preconditions"/>
+            <property name="illegalPattern" value="true"/>
+            <property name="message"
+                      value="Do not use Guava's Preconditions directly"/>
```
Comment (zhoujinsong) [inline tools/maven/checkstyle.xml:120]: Can we add some guidance in the message to instruct developers on how to solve this style error?

File: tools/maven/suppressions.xml [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+specific language governing permissions and limitations
+under the License.
+-->
+
+<!DOCTYPE suppressions PUBLIC
+		"-//Puppy Crawl//DTD Suppressions 1.1//EN"
+		"http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+		<suppress
```
Comment (zhoujinsong) [inline tools/maven/suppressions.xml:26]: We may fix check style errors in these files in the next PRs.

**Review by zhoujinsong — COMMENTED** [review]
Thanks a lot for pushing forward this improvement!
The new checkstyle improvements have enhanced existing code and reduced the workload for code reviewers.

I left some comments, PTAL.

File: tools/maven/suppressions.xml (hunk 3 above)
Comment (klion26) [reply tools/maven/suppressions.xml:26]: filed an issue #2675 to track this

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Comment by klion26** [comment]
@zhoujinsong thanks for the review and merging!

**Comment by zhoujinsong** [comment]
@klion26 I found that the current check style rules only check imports for classes under `com.google.common.base.Preconditions ` in Guava. Should we extend the checking rules to the entire Guava library like all classes under `com.google.common`?

**Comment by klion26** [comment]
@zhoujinsong this has been improved in #2679

**Comment by zhoujinsong** [comment]
> @zhoujinsong this has been improved in #2679

I got it. Thanks a lot for your work.

**Pull request merged**